lib/Conversion/ImportVerilog/FormatStrings.cpp

```diff
@@ -207,6 +210,36 @@ struct FormatStringParser {
     return success();
   }
 
+  // Format an integer with the %t specifier according to IEEE 1800-2023
+  // § 20.4.3 "$timeformat"
+  LogicalResult emitTime(const slang::ast::Expression &arg,
+                         const FormatOptions &options) {
+
+    // Only handle `TimeType` values.
+    auto value = context.convertRvalueExpression(
+        arg, moore::TimeType::get(context.getContext()));
+    if (!value)
+      return failure();
```

Comment:
Nice! I think this will already handle integers properly, too. The conversion will insert casts from integers to the time type 🚀